ibis/tests/expr/test_table.py

```diff
@@ -1495,6 +1496,32 @@ def test_unbound_table_name():
     assert match is not None
 
 
+class MyTable:
+    a: int
+    b: str
+    c: List[float]
+
+
+@ibis.table
+class MyDecoratedTable(MyTable):
+    pass
+
+
+def test_unbound_table_using_class_definition():
+    expected_schema = ibis.schema({'a': 'int64', 'b': 'string', 'c': 'array<double>'})
+
+    t1 = ibis.table(MyTable)
+    t2 = ibis.table(MyTable, name="MyNamedTable")
+    t3 = MyDecoratedTable
+
+    cases = {t1: "MyTable", t2: "MyNamedTable", t3: "MyDecoratedTable"}
```

Comment:
I don't think we need this test case - it's identical to `t1`, only the syntax used to make the call is different. The decorator syntax is a bit confusing IMO, since the return value of the decorator is still an `UnboundTable` instance rather than a type (class decorators usually return a type, not an instance).

```suggestion
class MyTable:
    a: int
    b: str
    c: List[float]


def test_unbound_table_using_class_definition():
    expected_schema = ibis.schema({'a': 'int64', 'b': 'string', 'c': 'array<double>'})

    t1 = ibis.table(MyTable)
    t2 = ibis.table(MyTable, name="MyNamedTable")

    cases = {t1: "MyTable", t2: "MyNamedTable"}
```

Reply:
SGTM, updating.